dagstore.go

```diff
@@ -50,22 +50,41 @@ type DAGStore struct {
 	indices index.FullIndexRepo
 	store   ds.Datastore
 
+	// Channels owned by us.
+	//
 	// externalCh receives external tasks.
 	externalCh chan *task
 	// internalCh receives internal tasks to the event loop.
 	internalCh chan *task
 	// completionCh receives tasks queued up as a result of async completions.
 	completionCh chan *task
-	// dispatchCh is serviced by the dispatcher goroutine.
-	dispatchCh chan *dispatch
+	// dispatchResultsCh is a buffered channel for dispatching results back to
+	// the application. Serviced by a dispatcher goroutine.
+	// Note: This pattern decouples the event loop from the application, so a
+	// failure to consume immediately won't block the event loop.
```

Comment:
However, slow consumption by consumers can block the event loop if the dispatch channel gets full.